src/spark_rapids_ml/classification.py

```diff
@@ -157,6 +182,39 @@ def __init__(
         )
         self._num_classes = num_classes
 
+    def _get_cuml_transform_func(
+        self, dataset: DataFrame
+    ) -> Tuple[
+        Callable[..., CumlT],
+        Callable[[CumlT, Union[cudf.DataFrame, np.ndarray]], pd.DataFrame],
+    ]:
+        _construct_rf, _ = super()._get_cuml_transform_func(dataset)
+
+        def _predict(rf: CumlT, pdf: Union[cudf.DataFrame, np.ndarray]) -> pd.Series:
+            data = {}
+            rf.update_labels = False
+            data[pred.prediction] = rf.predict(pdf)
+            data[pred.probability] = pd.Series(list(rf.predict_proba(pdf)))
+            return pd.DataFrame(data)
+
+        return _construct_rf, _predict
+
+    def _handle_multi_predication(
+        self, dataset: DataFrame, pred_struct_col_name: str
+    ) -> DataFrame:
+        probability_col = self.getProbabilityCol()
+        dataset = dataset.withColumn(
+            probability_col,
+            array_to_vector(getattr(col(pred_struct_col_name), pred.probability)),
+        )
+        return dataset
+
+    def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
```

Comment:
Can we simplify this by using final column names in _out_schema and assume that any array type columns will be converted to Vector type?   This could eliminate need for separate handle_multi_prediction method that has been added.

Reply:
What do you mean "using final column names in _out_schema"?

just like the following code without type information?

``` python
    def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
          schema = "prediction, probability"
        return schema
```

If so, spark will throw an exception when parsing the schema

``` console
E               pyspark.sql.utils.ParseException: 
E               Syntax error at or near ',': extra input ','(line 1, pos 10)
E               
E               == SQL ==
E               prediction, probability
E               ----------^^^
E               
E               
E               JVM stacktrace:
E               org.apache.spark.sql.catalyst.parser.ParseException: 
E               Syntax error at or near ',': extra input ','(line 1, pos 10)
E               
E               == SQL ==
E               prediction, probability
E               ----------^^^

```

Reply:
I mean use the names specified by probabilityCol and predictionCol instead of the temporary placeholder names prediction and probability.

Reply:
@wbo4958 ^^ .  If above is possible, _handle_multi_predication may not be needed.

Reply:
Done. Thx

Reply:
Did you change anything re. above?

Reply:
Oops, I just forget to push the new change. Now it's there.